biosdk-services/src/main/java/io/mosip/biosdk/services/utils/Utils.java

```diff
@@ -26,7 +41,175 @@ public RequestDto getRequestInfo(String request) throws ParseException {
+		stringBuilder.append(matchRequestDto.getModalitiesToMatch());
+		stringBuilder.append(", sample: ");
+		appendString(matchRequestDto.getSample(), stringBuilder);
+		stringBuilder.append(", gallery: ");
+		appendString(Arrays.stream(matchRequestDto.getGallery()).iterator(), stringBuilder, Utils::appendString);
+		stringBuilder.append(" }");
+		return stringBuilder.toString();
+	}
+
+	public static String toString(InitRequestDto initRequestDto) {
+		if(initRequestDto == null) {
+			return "null";
+		}
+		StringBuilder stringBuilder = new StringBuilder();
+		stringBuilder.append("InitRequestDto: { ");
+		stringBuilder.append("initParams:");
+		stringBuilder.append(" }");
+		return stringBuilder.toString();
+	}
+
+	public static String toString(CheckQualityRequestDto checkQualityRequestDto) {
+		if(checkQualityRequestDto == null) {
+			return "null";
+		}
+		StringBuilder stringBuilder = new StringBuilder();
+		stringBuilder.append("CheckQualityRequestDto: { ");
+		stringBuilder.append("flags:");
+		stringBuilder.append(checkQualityRequestDto.getFlags());
+		stringBuilder.append(", modalitiesToCheck: ");
+		stringBuilder.append(checkQualityRequestDto.getModalitiesToCheck());
+		stringBuilder.append(", sample: ");
+		appendString(checkQualityRequestDto.getSample(), stringBuilder);
+		stringBuilder.append(" }");
+		return stringBuilder.toString();
+	}
+
+	public static String toString(SegmentRequestDto segmentRequestDto) {
+		if(segmentRequestDto == null) {
+			return "null";
+		}
+		StringBuilder stringBuilder = new StringBuilder();
+		stringBuilder.append("SegmentRequestDto: { ");
+		stringBuilder.append("flags:");
+		stringBuilder.append(segmentRequestDto.getFlags());
+		stringBuilder.append(", modalitiesToSegment: ");
+		stringBuilder.append(segmentRequestDto.getModalitiesToSegment());
+		stringBuilder.append(", sample: ");
+		appendString(segmentRequestDto.getSample(), stringBuilder);
+		stringBuilder.append(" }");
+		return stringBuilder.toString();
+	}
+
+	public static String toString(ConvertFormatRequestDto convertFormatRequestDto) {
+		if(convertFormatRequestDto == null) {
+			return "null";
+		}
+		StringBuilder stringBuilder = new StringBuilder();
+		stringBuilder.append("ConvertFormatRequestDto: { ");
+		stringBuilder.append("sourceFormat:");
+		stringBuilder.append(convertFormatRequestDto.getSourceFormat());
+		stringBuilder.append(", targetFormat: ");
+		stringBuilder.append(convertFormatRequestDto.getTargetFormat());
+		stringBuilder.append(", modalitiesToConvert: ");
+		stringBuilder.append(convertFormatRequestDto.getModalitiesToConvert());
+		stringBuilder.append(", sample: ");
+		stringBuilder.append(", sourceParams:");
+		stringBuilder.append(convertFormatRequestDto.getSourceParams());
+		stringBuilder.append(", targetParams: ");
+		stringBuilder.append(convertFormatRequestDto.getTargetParams());
+		appendString(convertFormatRequestDto.getSample(), stringBuilder);
+		stringBuilder.append(" }");
+		return stringBuilder.toString();
```

Comment:
why there is so many append string, why can't we use the some object mapper or json convertor whichever give better performance ?

Reply:
The problem with object mapper was with replacing the BDB value with its hash value cannot be achieved. Also it would be complex to replace it after the conversion happened if we use the such converter. So preferred to use the StringBuilder.

Reply:
Hi @vishwa-vyom , I explored such option (https://stackoverflow.com/questions/56263350/lombok-excluding-field-with-tostring-exclude-is-not-working). But since this needs change in DTO which resides in commons kernel-biometrics-api library which we should avoid, and also there is possibility for the this library being overridden by any older version by any other dependency which would result in accidental exposure of the BDB data.